File: backend/app/models/audit.py
from datetime import datetime

from sqlalchemy import DateTime, Integer, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class AuditLog(Base):
    __tablename__ = "audit_logs"

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    identity_id: Mapped[int | None] = mapped_column(Integer, nullable=True)
    identity_type: Mapped[str] = mapped_column(String(20))
    action: Mapped[str] = mapped_column(String(50))
    resource: Mapped[str] = mapped_column(String(200))
    request_summary: Mapped[str | None] = mapped_column(Text, nullable=True)
    response_summary: Mapped[str | None] = mapped_column(Text, nullable=True)
    ip: Mapped[str | None] = mapped_column(String(50), nullable=True)
    duration_ms: Mapped[int | None] = mapped_column(Integer, nullable=True)
    status: Mapped[str] = mapped_column(String(20))  # success/error
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now()
    )

File: frontend/src/api/audit.ts
import request from './index'

export interface AuditLog {
  id: number
  identity_id: number | null
  identity_type: string
  username: string
  action: string
  resource: string
  request_summary?: string
  response_summary?: string
  ip?: string
  duration_ms?: number
  status: string
  created_at: string
}

export interface AuditQuery {
  page?: number
  page_size?: number
  identity_id?: number
  identity_type?: string
  action?: string
}

export interface PaginatedResult<T> {
  items: T[]
  total: number
  page: number
  page_size: number
}

export function getAuditLogs(params: AuditQuery): Promise<PaginatedResult<AuditLog>> {
  return request.get('/audit/logs', { params })
}
